newIDE/app/src/GameDashboard/GameRegistration.js

```diff
@@ -107,42 +122,67 @@ export const GameRegistration = ({
         if (onGameRegistered) onGameRegistered();
       } catch (error) {
         console.error('Unable to register the game', error);
-        showErrorBox({
-          rawError: error,
-          errorId: 'register-game-error',
-          // TODO: i18n
-          message:
-            'Unable to register the game.' +
-            ' ' +
-            'Verify your internet connection or try again later.',
+        showAlert({
+          title: t`Unable to register the game`,
+          message: t`Verify your internet connection or try again later.`,
         });
       }
       setRegistrationInProgress(false);
     },
-    [getAuthorizationHeader, profile, project, loadGame, onGameRegistered]
+    [
+      getAuthorizationHeader,
+      profile,
+      project,
+      loadGame,
+      onGameRegistered,
+      showAlert,
+    ]
   );
 
-  const _onAcceptGameStatsEmail = React.useCallback(
-    async () => {
-      if (!profile || !project) return;
+  const onToggleGameStatsEmail = React.useCallback(
+    async (value: boolean) => {
+      if (!profile || !game) return;
+
+      setToggleGameStatsEmailInProgress(true);
+      try {
+        await onAcceptGameStatsEmail(value);
+      } catch (error) {
+        console.error('Unable to toggle game stats email.', error);
+        showAlert({
+          title: t`Unable to toggle game stats email`,
```

Comment:
Unable to toggle is technical jargon. Prefer "Unable to change your email preferences"